code/modules/antagonists/eldritch_cult/knowledge.dm/ash_lore.dm

```diff
@@ -0,0 +1,202 @@
+	name = "Ashen Eyes"
+	gain_text = "Piercing eyes may guide me through the mundane."
+	desc = "Allows you to craft thermal vision amulet by transmutating eyes with a glass shard."
+	cost = 1
+	next_knowledge = list(/datum/eldritch_knowledge/spell/ashen_shift,/datum/eldritch_knowledge/flesh_ghoul)
+	required_atoms = list(/obj/item/organ/eyes,/obj/item/shard)
+	result_atoms = list(/obj/item/clothing/neck/eldritch_amulet)
+
+/datum/eldritch_knowledge/ash_mark
+	name = "Mark of ash"
+	gain_text = "Spread the famine."
+	desc = "Your sickly blade now applies ash mark on hit. Use your mansus grasp to proc the mark. Mark of Ash causes stamina damage, and fire loss, and spreads to a nearby carbon. Damage scales with how many times the mark has spread."
+	cost = 2
+	next_knowledge = list(/datum/eldritch_knowledge/curse/blindness)
+	banned_knowledge = list(/datum/eldritch_knowledge/rust_mark,/datum/eldritch_knowledge/flesh_mark)
+	route = "Ash"
+
+/datum/eldritch_knowledge/ash_mark/eldritch_blade_act(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(istype(target,/mob/living))
+		var/mob/living/L = target
+		L.apply_status_effect(/datum/status_effect/eldritch/ash,1)
+
+/datum/eldritch_knowledge/curse/blindness
+	name = "Curse of blindness"
+	gain_text = "Blind man walks through the world, unnoticed by the masses."
+	desc = "Curse someone with 5 minutes of complete blindness."
+	cost = 1
+	required_atoms = list(/obj/item/organ/eyes,/obj/item/kitchen/knife,/obj/effect/decal/cleanable/blood)
+	next_knowledge = list(/datum/eldritch_knowledge/curse/corrosion,/datum/eldritch_knowledge/ash_blade_upgrade,/datum/eldritch_knowledge/curse/paralysis)
+	timer = 2 MINUTES
+	route = "Ash"
+
+/datum/eldritch_knowledge/curse/blindness/curse(mob/living/chosen_mob)
+	. = ..()
+	chosen_mob.become_blind(MAGIC_TRAIT)
+
+/datum/eldritch_knowledge/curse/blindness/uncurse(mob/living/chosen_mob)
+	. = ..()
+	chosen_mob.cure_blind(MAGIC_TRAIT)
+
+/datum/eldritch_knowledge/spell/mad_touch
+	name = "Touch of Madness"
+	gain_text = "Take what's yours."
+	desc = "You gain a spell that destroys enemies sanity."
+	cost = 1
+	spell_to_add = /obj/effect/proc_holder/spell/targeted/touch/mad_touch
+	next_knowledge = list(/datum/eldritch_knowledge/spell/blood_siphon,/datum/eldritch_knowledge/summon/ashy,/datum/eldritch_knowledge/ash_final)
+	route = "Ash"
+
+/datum/eldritch_knowledge/ash_blade_upgrade
+	name = "Fiery blade"
+	gain_text = "May the sun burn the heretics."
+	desc = "Your blade of choice will now add firestacks"
+	cost = 2
+	next_knowledge = list(/datum/eldritch_knowledge/spell/mad_touch)
+	banned_knowledge = list(/datum/eldritch_knowledge/rust_blade_upgrade,/datum/eldritch_knowledge/flesh_blade_upgrade)
+	route = "Ash"
+
+/datum/eldritch_knowledge/ash_blade_upgrade/eldritch_blade_act(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.IgniteMob()
+		C.adjust_fire_stacks(1)
```

Comment:
the IgniteMob() needs to come AFTER the adjustment of their fire stacks

also, adding only one fire stack per hit is kind of pathetic, and this is also gonna set YOU on fire as well

as I said before, adding a fire immunity passive upgrade to the ashie tech tree would be really nice

Reply:
oh aight i will fix it